src/util/db/dbconnectionpool.cpp

```diff
@@ -0,0 +1,92 @@
+#include "util/db/dbconnectionpool.h"
+
+#include "util/memory.h"
+#include "util/logger.h"
+
+
+namespace mixxx {
+
+namespace {
+
+const mixxx::Logger kLogger("DbConnectionPool");
+
+} // anonymous namespace
+
+bool DbConnectionPool::createThreadLocalConnection() {
+    VERIFY_OR_DEBUG_ASSERT(!m_threadLocalConnections.hasLocalData()) {
+        DEBUG_ASSERT(m_threadLocalConnections.localData());
+        kLogger.critical()
+                << "Thread-local database connection already exists"
+                << *m_threadLocalConnections.localData();
+        return false; // abort
+    }
+    const int connectionIndex =
+            m_connectionCounter.fetchAndAddAcquire(1) + 1;
+    const QString indexedConnectionName =
+            QString("%1-%2").arg(
+                    m_prototypeConnection.name(),
+                    QString::number(connectionIndex));
+    auto pConnection = std::make_unique<DbConnection>(m_prototypeConnection, indexedConnectionName);
+    if (!pConnection->open()) {
+        kLogger.critical()
+                << "Failed to open thread-local database connection"
+                << *pConnection;
+        return false; // abort
+    }
+    m_threadLocalConnections.setLocalData(pConnection.get()); // transfer ownership
+    pConnection.release(); // release ownership
+    DEBUG_ASSERT(m_threadLocalConnections.hasLocalData());
+    DEBUG_ASSERT(m_threadLocalConnections.localData());
+    kLogger.info()
+            << "Cloned thread-local database connection"
+            << *m_threadLocalConnections.localData();
+    return true;
+}
+
+void DbConnectionPool::destroyThreadLocalConnection() {
+    m_threadLocalConnections.setLocalData(nullptr);
+}
+
+QSqlDatabase DbConnectionPool::threadLocalConnection() const {
```

Comment:
Thank you for your good explanations, I think I understand now. 
The source of confusion has to do with this function because we have now two owner of the connection. 
On one hand the scoped connection and on the other hand the connection pool. 

I see two possible solutions: 
1. Remove this function here and move it into the scoped connection. Than the caller is instantly aware who actually owns the connection 
2. Keep this function and hand the connection out as a shared pointer. Remove the ThreadLocalScoped class. 
Than the connection pools owns the connection and actually just one for one thread.